src/main/resources/data/minecoprocessors/books/manual.txt

```diff
@@ -1,130 +1,291 @@
 Redstone Processor Manual
 Minecoprocessors
+§r
+§r
+§r   Minecoprocessors
+§r
+§r        Redstone
+§r       Processor
+§r        Manual
+§r
+§r
+§r
+§oVisit the Wiki§r §1https://github.com/ToroCraft/Minecoprocessors/wiki§r
+~~~
+§nIndex§r
+
+The first pages of this manual are dense "cheat sheets". A more detailed introduction starts at page §l10§r.
+
+Some examples are attached as appendix.
+
+
+
+~~~
+§nI/O Registers§r
+
+§1pf, pb, pl, pr§r: I/Os
+
+§1ports§r: I/O config
+§1adc§r: ADC config
+§1idr§r: Input data reg
+§1odr§r: Output data reg
+
+Bit order for §1ports§r, §1adc§r, §1idr§r, §1odr§r: front (0001b), back (0010b), left (0100b), right (1000b)§r.
+~~~
+§nBasic Instructions§r
+
+§1MOV a, b§r Move/assign
+§1NOP§r No Operation
+§1CLC§r Clear Carry
+§1SEC§r Set Carry
+§1CLZ§r Clear Zero
+§1SEZ§r Set Zero
+
+§1PUSH a§r Push to Stack
+§1POP a§r Pop from Stack
+
+§1WFE§r Wait For Event
+§1HLT§r Halt Processor
+~~~
+§nArithmetic Instructions§r
+
+§1ADD a, b§r Add
+§1SUB a, b§r Subtract
+§1MUL a§r Multiply
+§1DIV a§r Divide
+§1INC a§r Increment by 1
+§1DEC a§r Decrement by 1
+
+
+
+
+
+§o(Results stored in §1a§r§o).§r
+~~~
+§nLogical Instructions§r
+
+§1AND a, b§r Bitwise AND
+§1OR a, b§r Bitwise OR
+§1XOR a, b§r Bitwise XOR
+§1NOT a§r Bitwise NOT
+
+
+
+
+
+
+
+§o(Results stored in §1a§r§o).§r
+~~~
+§nShift Instructions§r
+
+§1SHL a§r Shift Left
+§1SHR a§r Shift Right
+§1SAR a§r Arithmetic Right
+§1SAL a§r Arithmetic Left
+§1ROR a§r Rotate Right
+§1ROL a§r Rotate Right
+
+
+
+
+
+§o(Results stored in §1a§r§o).§r
+~~~
+§nFlow Control Instr.§r
+
+§1CMP a, b§r Compare
+§1JMP label§r Jump
+§1JZ label§r Jump if Zero
+§1JNZ label§r not Zero
+§1JC label§r if Carry
+§1JNC label§r if no Carry
+§1DJNZ a, label§r Dec and Jump if not zero
+§1CALL label§r Call
+§1RET§r Return
+~~~
+§nFlags§r
+
+§1Z§r Zero Flag
+§1C§r Carry Flag
+§1F§r Fault Flag
+§1S§r Sleep Flag
+
+
+
 
-  Minecoprocessors
 
-       Redstone
-      Processor
-        Manual
 
 ~~~
-§1Visit the Wiki§r
 
-For more information, visit the wiki:
 
-https://github.com/ToroCraft/Minecoprocessors/wiki
+
+     §o§lIntroduction§r
+
+~~~
+§nSyntax§r
+
+In assembly language we do one operation per line. Each line starts with an instruction, followed by arguments, e.g.:
+
+ §1MOV a, 10§r
+
+assigns 10 to the register §1a§r ("§1a = 10§r").
+~~~
+Instructions which calculate something normally store the result back in the first argument, hence,
+
+ §1MOV a, 12§r
+ §1MOV b, 10§r
+ §1SUB a, b§r
+
+will result in register §1a§r containing §12§r as present value.
```

Comment:
I'd reccommend changing this to read
`will result in register §1a§r containing the value §12§r.`